### PR TITLE
[MRG] fix a bug in gather when high-resolution queries are used

### LINKED ISSUE #398: gather output confusion
Hi,

Have been playing around with sourmash and it seems to be quite a handy tool. However, I am having trouble making sense of the output data.

I ran sourmash gather with a metagenome assembled genome (MAG) of 5.8 Mb against the GenBank SBT files provided as a prepared database (https://s3-us-west-1.amazonaws.com/spacegraphcats.ucdavis.edu/microbe-genbank-sbt-k31-2017.05.09.tar.gz).

Then, in my report I get:

```
overlap     p_query p_match 
---------   ------- --------
5.7 Mbp     100.0%  100.0%      CTEK01000001.1 Paenibacillus sp. GM2 ...

found 1 matches total;
the recovered matches hit 50.1% of the query
```

There are two things that I did not quite understand: one is that although I get 100% p_query and p_match, it only says that I recovered matches for 50% of the query; additionally, although the overlap is 5.7 Mb (which is almost 100% of my MAG at 5.8 Mb) the reference match CTEK01000001.1 is actually one contig of 800 bp (https://www.ncbi.nlm.nih.gov/nuccore/CTEK01000001.1).

Would you be able to help me better understand what is going on?

Many thanks in advance,
Alex

### PR DESCRIPTION
**Pull request by ctb**
Fixes #398.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review @luizirber @brooksph @taylorreiter

**Comment by ctb** [comment]
bump!

File: sourmash_lib/search.py [hunk 1]
```diff
@@ -162,13 +163,16 @@ def build_new_signature(mins, template_sig):
...
-        # CTB: these could probably be replaced by minhash.downsample_scaled.
-        new_max_hash = sourmash_lib.MAX_HASH / float(R_comparison)
+        # eliminate mins under this new resolution.
+        # (CTB note: this means that if a high scaled/low res signature is
+        # found early on, resolution will be low from then on.)
+        new_max_hash = get_max_hash_for_scaled(R_comparison)
         query_mins = set([ i for i in query_mins if i < new_max_hash ])
         found_mins = set([ i for i in found_mins if i < new_max_hash ])
         orig_mins = set([ i for i in orig_mins if i < new_max_hash ])
+        sum_abunds = sum([ v for (k,v) in orig_abunds.items() if k < new_max_hash ])
```
Comment (taylorreiter) [inline sourmash_lib/search.py]: I don't understand why not having this in previously didn't cause a problem, but I understand why it should be present

**Comment by brooksph** [comment]
@taylorreiter @ctb LGTM

**Comment by ctb** [comment]
thank you both!

**Comment by luizirber** [comment]
For future ref, let's try to keep cleaning up/reformatting in separate PRs, it is harder to review if there are unrelated changes ]

**Comment by ctb** [comment]
-0 from my perspective; the test was pretty clear, and a lot of the
"grooming" of the code happened integrally with tracking down the
bug.

I am wary of having more long-running PRs around, as well. Maybe they're
long running because they're hard to review, but that is not my guess :)

*shrug* those are my thoughts :)

**Pull request merged**